### PR TITLE
scripts: runners: spi_burn: replaced delay by wait for response

### PR DESCRIPTION
**Pull request by yurvyn**
Replaced hard-coded delay by wait for response during ICEman
connection.

Signed-off-by: Yuriy Vynnychek <yura.vynnychek@telink-semi.com>

### CODE REVIEW

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
No objections to the approach, just some requests about the specifics. Thanks for the patch.

**Comment by rikorsev** [comment]
Hi @mbolivar-nordic 

Thanks for review. Requested changes are done. Please approve if everything is fine now.

File: scripts/west_commands/runners/spi_burn.py
Comment (mbolivar-nordic) [inline scripts/west_commands/runners/spi_burn.py:87]: In the latest round of changes, you moved the only uses of these values back into the `_flash` method. They are not used outside of the method anymore. Therefore, there is no reason to store them as instance attributes anymore either. Please keep these as local variables.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Another cleanup requested due to the way the latest push was done.

File: scripts/west_commands/runners/spi_burn.py
Comment (rikorsev) [reply scripts/west_commands/runners/spi_burn.py]: Done

File: scripts/west_commands/runners/spi_burn.py
Comment (mbolivar-nordic) [inline scripts/west_commands/runners/spi_burn.py:111]: Why was this added in the latest push? Please remove it; we don't leave debug printlines around like this.

Comment (rikorsev) [reply scripts/west_commands/runners/spi_burn.py]: Removed

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
Thanks @yurvyn -- can you please suggest some other reviewers if you don't get another +1 soon?

**Comment by yurvyn** [comment]
@carlescufi,
One more approval is required.
Could you please approve the PR?

**Comment by yurvyn** [comment]
@swinslow, @carlescufi 

One more approval is required.
Could you please approve the PR?

**Comment by swinslow** [comment]
Hi @yurvyn, unfortunately my only familiarity in the Zephyr scripts is for the [SPDX generator code](https://github.com/zephyrproject-rtos/zephyr/tree/main/scripts/west_commands/zspdx) -- so I'm not sure I can say anything meaningful in reviewing this PR.

Is there anyone else in the Zephyr community who might be using or needing this, who you could suggest as a reviewer?

**Comment by rikorsev** [comment]
Hi @mbolivar-nordic, 

The review process is stall due to absence of reviewers who can review the PR. Could you assign someone to review the PR or just merge it?

**Comment by yurvyn** [comment]
@nandojve @nashif @sylvioalves @henrikbrixandersen

Please provide the last required approval.

Thanks,
Yuriy

**Review by carlescufi — APPROVED** [review]
LGTM

**Pull request merged**
